ui-frontend/packages/catalog-ui-search/src/main/webapp/component/search-form-interactions/search-form-interactions.view.js

```diff
@@ -128,6 +139,44 @@ Anyone who has access to this search ${formTitleLowerCase} will subsequently los
       !user.canShare(this.model)
     )
     this.$el.toggleClass('is-not-editable-template', !user.canWrite(this.model))
+    this.$el.toggleClass(
+      'is-not-shared-template',
+      this.model.get('createdBy') === 'system' ||
+        this.model.get('createdBy') === user.getUserId() ||
+        user.hasGroupAccess(this.model)
+    )
+  },
+  handleRemoveSharedForm() {
+    this.listenTo(
+      ConfirmationView.generateConfirmation({
+        prompt: `This will permanently remove your access to the shared ${formTitleLowerCase}. Would you like to continue?`,
+        no: 'Cancel',
+        yes: 'Remove',
+      }),
+      'change:choice',
+      confirmation => {
+        if (confirmation.get('choice')) {
+          let loadingview = new LoadingView()
+          const id = this.model.get('id')
+          handleRemoveSharedMetacard(id).then(res => {
+            if (res.status !== 200) {
+              announcement.announce(
+                {
+                  title: 'Error',
+                  message: `Unable to leave the ${formsTitleLowerCase}`,
```

Comment:
❓ What will this error message end up being? `Unable to leave the SearchFormName`? Should it be something more like `Unable to leave the SearchFormName sharing list`?